Network/HTTP2/H2/Manager.hs

```diff
@@ -117,10 +143,9 @@ waitCounter0 (Manager _timmgr var) = atomically $ do
 
 ----------------------------------------------------------------
 
-withTimeout :: Manager -> (T.Handle -> IO ()) -> IO ()
-withTimeout (Manager timmgr var) action =
-    T.withHandleKillThread timmgr (return ()) $ \th -> do
-        tid <- myThreadId
-        -- overriding ThreadWithoutTimeout
-        atomically $ modifyTVar var $ Map.insert tid $ ThreadWithTimeout th
-        action th
+myWeakThradId :: IO (Weak ThreadId, Int)
+myWeakThradId = do
+    tid <- myThreadId
+    wtid <- mkWeakThreadId tid
+    let n = read (drop 9 $ show tid) -- drop "ThreadId "
```

Comment:
@kazu-yamamoto I am late to the party but there’s fromThreadId in base. It allows you to get a stable integer identifier of a thread without resorting to the show instance.

Reply:
I cannot find `fromThreadId` anywhere?
Which module exports it?

Reply:
https://hackage.haskell.org/package/base-4.20.0.1/docs/GHC-Conc-Sync.html#v:fromThreadId

Reply:
`fromThreadId` is very new, sigh.

Reply:
FWIW, I don't think the weak references are strictly necessary. Provided that we keep the `Map` in WHNF, and we remove entries whenever the corresponding threads die, we should not be unnecessarily retaining any threads.